src/main/java/fr/ens/biologie/genomique/eoulsan/modules/diffana/DESeq2Checker.java

```diff
@@ -79,39 +118,37 @@ static boolean checkExperimentDesign(Experiment experiment, boolean throwsExcept
                     }
                 }
             }
-        }
-
-        // Check if the column Condition is missing for the experiment
-        if (!getExperimentSampleAllMetadataKeys(experiment)
-                .contains(CONDITION_KEY)
-                && !getAllSamplesMetadataKeys(experiment.getDesign()).contains(CONDITION_KEY)) {
-            return error("Condition column missing for experiment: "
-                    + experiment.getName(), throwsException);
-        }
-
-        // Check if the column RepTechGroup is missing for the experiment
-        if (!getExperimentSampleAllMetadataKeys(experiment)
-                .contains(REP_TECH_GROUP_KEY)
-                && !getAllSamplesMetadataKeys(experiment.getDesign()).contains(REP_TECH_GROUP_KEY)) {
-            return error("RepTechGroup column missing for experiment: "
-                    + experiment.getName(), throwsException);
+            // Check if the name of each comparison is unique
+            Set<String> set=new HashSet<>();
+            Set<String> duplicateElements=new HashSet<>();
+            for (String element : comparisionNames) {
+                if(!set.add(element)){
+                    duplicateElements.add(element);
+                }
+            }
```

Comment:
It is easier to declare a duplicateElements set and fill it in the `for (String string : splitC)` loop.